.codex/skills/construction-contract-analysis/scripts/agnes_mcp_server.py

```
@@ -0,0 +1,150 @@
+            bitmap = page.render(scale=2.0)
+            img = bitmap.to_pil()
+            buf = io.BytesIO()
+            img.save(buf, format='JPEG', quality=80)
+            b64 = base64.b64encode(buf.getvalue()).decode()
+
+            result = _vision_ocr(b64, prompt)
+            if 'choices' in result:
+                output.append(result['choices'][0]['message']['content'])
+            else:
+                output.append(f"失败: {result.get('error', 'Unknown')}")
+        except Exception as e:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)